Python/Module5_OddsAndEnds/WorkingWithFiles.md

```diff
@@ -249,6 +249,102 @@ with open("a_poem.txt", mode="r") as my_open_file:
 ```
 <!-- #endregion -->
 
+<!-- #region -->
+## Working with Comma Seperated Value Files
+
+Comma Seperated Value (CSV) files are commonly used to store data that you might typically find in a table. These files can be formatted in many ways, but the typical format is to have each of the column values in the table be separated by commas while having a newline separate each row. Suppose we have the following table of test scores:
+
+|         | Exam 1 (%)           | Exam 2 (%) |
+| ------------- |:-------------:| -----:|
+| Ashley     | $93$ | $95$ |
+| Brad     | $84$      |   $100$ |
+| Cassie | $99$      |    $87$ |
+
+This table, which sourced for Module 3 of Python Like You Mean It, depicts the test scores of three students across 2 exams. Here is what the corresponding CSV file might look like:
+
+```python
+name,exam one score,exam two score
+Ashley,93,95
+Brad,84,100
+Cassie,99,87
+```
+In addition to the fact that the first line typically contains headers columns, you are also allowed to have spaces within each of columns as well.
+
+<div class="alert alert-warning">
+
+**Note**: 
+
+It is not guaranteed that all CSV files are actually comma separated. Non-standard CSV files will typically come with instructions on how the data is organized. In general, it is a good practice to open up the CSV file (using Notepad or a regular text editor) and look at the first few lines to get a sense of how it is organized (unless the file is too large).
+</div>
+
+### How to parse CSVs with Numpy
+
+We will first look into parsing and storing CSV data using our favorite package: Numpy!
+```python
+from numpy import genfromtxt # genfromtxt() allows for easy parsing of CSVs
+my_data = genfromtxt(r"path_to_csv/file.csv", delimiter=',') 
+```
+`genfromtxt()` takes in CSV file path and delimiter (what is used to split the data, typically comma for CSV).
+Let's check some properties:
+```python
+>>> type(my_data)
+numpy.ndarray
+
+>>> my_data.shape
+(numSamples, numAttributesPerSample)
+
+#Let's look at the actual data
+>>> my_data
+array([[    nan,     nan,     nan, ...,     nan,     nan,     nan],
+       [    nan, -99.9  , -99.9  , ..., -99.9  , -99.9  , -99.9  ],
+       [    nan,   0.875,   1.39 , ...,   4.506, -99.9  , -99.9  ],
+       ...,
+       [    nan,   2.157,   3.43 , ...,  12.89 ,  97.   ,  21.95 ],
+       [    nan,   2.087,   2.84 , ...,  10.963,  92.   ,  21.95 ],
+       [    nan,   1.926,   2.98 , ...,  12.228,  84.   ,  21.95 ]])
+```
```

Comment:
In general, we usually have code blocks that a reader can actually run. Would it be possible to make the CSV available and have actual load commands? Then you shouldn't have `(numSamples, numAttributesPerSample)` as the shape of `my_data`. Also.....`pleaseDoNotUseCamelCase`